dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -268,6 +265,7 @@ def load_level_source
         readonly_view_options
       end
       readonly_view_options if user_level && user_level.readonly_answers?
+      level_source = user_level.try(:level_source)
```

Comment:
It looks like @last_activity no longer gets set?

Also, this change will result in behavior changes - not just an optimization. Previously we would get the level_source for the last_activity by keyed off of level. With this change, we filter that only to levels in the given script.

I'm not sure we want to make this change yet. Although your change seems like the right behavior long term, we can't easily do the same thing with channel backed levels. We have an active thread with the education team talking about how we want to handle this.

Reply:
> It looks like @last_activity no longer gets set?

Thanks for pointing this out! I'll restore the instance variable. (The fact that this was able to be changed without breaking any tests suggests we're missing test coverage on some meaningful logic here..)

Reply:
> I'm not sure we want to make this change yet. Although your change seems like the right behavior long term, we can't easily do the same thing with channel backed levels. We have an active thread with the education team talking about how we want to handle this.

I'll jump on that thread. The single extra query providing the current behavior comprises somewhere around 3-5% of our total database activity, so there's a pretty high bar for not optimizing this.

Why can't we do the same thing with channel backed levels?

Reply:
Channels are tied to a particular level, rather than to a particular script_level, so if a channel-backed level is used on scriptA and scriptB, they both end up with the same channel.